lib/rubocop/cop/rspec/let_setup.rb

```diff
@@ -35,7 +35,10 @@ class LetSetup < Base
                          ).block_pattern
 
         def_node_matcher :let_bang, <<-PATTERN
-          (block $(send nil? :let! (sym $_)) args ...)
+          {
+            (block $(send nil? :let! {(sym $_) (str $_)}) ...)
+            $(send nil? :let! {(sym $_) (str $_)} block_pass)
+          }
```

Comment:
It might be possible to get rid of this complexity somehow by extracting a part of this. But refactoring can wait while we have more cases with block-pass vs regular.